substrate/frame/proxy/src/lib.rs

```diff
@@ -163,6 +163,9 @@ pub mod pallet {
 		/// into a pre-existing storage value.
 		#[pallet::constant]
 		type AnnouncementDepositFactor: Get<BalanceOf<Self>>;
+
+		/// Provider for the block number. Normally this is the `frame_system` pallet.
+		type BlockNumberProvider: BlockNumberProvider<BlockNumber = BlockNumberFor<Self>>;
```

Comment:
Why do we force the `BlockNumber` to be the frame system block number type?

Instead we should allow block number to be any type here and redefine a helper type:

```rust
type BlockNumberFor<T> = <T as Config>::BlockNumberProvider as BlockNumberProvider>::BlockNumber;
```
```suggestion
		type BlockNumberProvider: BlockNumberProvider;
```

Reply:
I agree, we should not force the same type. But probably we cannot change the `BlockNumberFor` type reference now, since not all configs have the `BlockNumberProvider`